src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebhookCreator.java

```diff
@@ -56,18 +51,10 @@ public static void register(SCMNavigatorOwner owner, GitLabSCMNavigator navigato
         }
         try {
             GitLabApi gitLabApi = new GitLabApi(server.getServerUrl(), credentials.getToken().getPlainText());
-            GitLabOwner gitLabOwner = GitLabOwner.fetchOwner(gitLabApi, navigator.getProjectOwner());
-            List<Project> projects;
-            LOGGER.info("Project Owner: "+ gitLabOwner);
-            if(gitLabOwner instanceof GitLabUser) {
-                Stream<Project> projectsStream = gitLabApi.getProjectApi().getOwnedProjectsStream();
-                projects = projectsStream
-                        .filter(project -> !project.getNamespace().getKind().equals("group"))
-                        .collect(Collectors.toList());
-            } else {
-                projects = gitLabApi.getGroupApi().getProjects(navigator.getProjectOwner());
+            List<Project> projects = new ArrayList<>();
+            for(String projectPathWithNamespace : navigator.getNavigatorProjects()) {
+                projects.add(gitLabApi.getProjectApi().getProject(projectPathWithNamespace));
```

Comment:
Sir, you can save API calls here as well :laughing: 

Something like this should work too :sweat_smile: 
```java
            HashSet<String> projects = navigator.getNavigatorProjects();
            if(projects.isEmpty()) {
                LOGGER.log(Level.WARNING,
                        "Group is empty!");
                return;
            }
            // Since GitLab doesn't allow API calls on Group WebHooks.
            // So fetching a list of web hooks in individual projects inside the group
            List<ProjectHook> validHooks = new ArrayList<>();
            // Filters all projectHooks and returns an empty Project Hook or valid project hook per project
            for(String p : projects) {
                validHooks.add(gitLabApi.getProjectApi().getHooksStream(p)
                        .filter(hook -> hookUrl.equals(hook.getUrl()))
                        .findFirst()
                        .orElse(new ProjectHook()));
            }
```

Reply:
Missed the part below:

```java
            List<String> projects = new ArrayList<>(navigator.getNavigatorProjects());
            if(projects.isEmpty()) {
                LOGGER.log(Level.WARNING,
                        "Group is empty!");
                return;
            }
            // Since GitLab doesn't allow API calls on Group WebHooks.
            // So fetching a list of web hooks in individual projects inside the group
            List<ProjectHook> validHooks = new ArrayList<>();
            // Filters all projectHooks and returns an empty Project Hook or valid project hook per project
            for(String p : projects) {
                validHooks.add(gitLabApi.getProjectApi().getHooksStream(p)
                        .filter(hook -> hookUrl.equals(hook.getUrl()))
                        .findFirst()
                        .orElse(new ProjectHook()));
            }
            for(ProjectHook hook : validHooks) {
                if(hook.getId() == null) {
                    String project = projects.get(validHooks.indexOf(hook));
                    ProjectHook enabledHooks = createHook();
                    // TODO add secret token, add more events give option for sslVerification
                    gitLabApi.getProjectApi().addHook(project, hookUrl, enabledHooks, false, "");
                }
            }
```